validator_option.go

```diff
@@ -0,0 +1,21 @@
+package jwt
+
+import "time"
+
+// ValidatorOption is used to implement functional-style options that modify the behavior of the parser. To add
+// new options, just create a function (ideally beginning with With or Without) that returns an anonymous function that
+// takes a *ValidatorOptions type as input and manipulates its configuration accordingly.
+type ValidatorOption func(*ValidatorOptions)
```

Comment:
maybe ValidationOption instead? Probably a toin coss at some point. But the way I see it, this function MUST be exported (or must it) so we should choose a name we are happy with.

cc @mfridman

Reply:
Only needs to be exported if we want to allow usage in the call to Valid outside of the parser.  Changed so nothing is exported except the `WithLeeway` function in the parser.

Reply:
Good idea. Probably the best approach until this is stabilized from an API point of view.